.github/workflows/ci.yml

```diff
@@ -41,10 +59,9 @@ jobs:
       run: |
         git clone https://github.com/nextstrain/zika-tutorial
         pip install nextstrain-cli
-        nextstrain build --image nextstrain/base:$TAG zika-tutorial -F
+        nextstrain build --image localhost:5000/nextstrain/base:$TAG zika-tutorial -F
 
```

Comment:
Now that CI succeeded and pushed [an arm64 image](https://hub.docker.com/layers/nextstrain/base/branch-victorlin-add-arm64-keep-scripts/images/sha256-352393ad341db327996d785778b01c4094c4555344225d09409cca422516e1ea), I pulled it locally on my M1 Mac for testing. It runs zika-tutorial fine but this error comes up when calling `nextalign` in the ncov workflow:

```
qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory
```

I need to revisit all the `x86`/`amd64` references in the Dockerfile and either (1) build them from source or (2) set up emulation within the Docker image.

Reply:
All `x86`/`amd64` references in [Dockerfile](https://github.com/nextstrain/docker-base/blob/6f97b669959dbb123cf58321386fc4492087ffe0/Dockerfile):

- `nextalign2`, `nextclade2`: calling the command alone results in error above.
- `nextalign1`, `nextclade1`: calling the command alone runs fine.
- `csvtk`: the [dim/nrow/ncol examples](https://bioinf.shenwei.me/csvtk/usage/#dimnrowncol) run fine.
- `tsv-utils`: the [`tsv-pretty` example](https://github.com/eBay/tsv-utils/blob/38ed0a1c31742bd8b59196517e89ff0b51e8fb80/README.md#tsv-pretty) runs fine.
- `gofasta`: untested
- `seqkit`: untested

My thought now is that this is specific to `nextclade2`/`nextalign2`.

Reply:
401506ed8d8dacf40f515d8ae9b4d45ef2b025ff should fix the error above, but still curious about emulation for the others.

Reply:
It seems like emulation is supported in the Docker image. This means `amd64` binaries can run on `arm64` without error, though it might be slower. I think emulation is acceptable if the tool cannot easily be built for the target platform, as long as emulation is noted in the Dockerfile.

I'll resolve this comment once each one of these references are addressed by either (1) finding the proper download, (2) trying to build from source, or (3) adding a comment noting on emulation.

Reply:
> It seems like emulation is supported in the Docker image. This means `amd64` binaries can run on `arm64` without error, though it might be slower.

Hmm. This feels artifactual to me. Like that it's a result of having enabled QEMU on the host system for building the image, or something similar. Knowing details about how Docker works, it's not clear to me how it would be generally supported by Docker on all platforms.

Are you testing in CI or on your own M1 system? Is this the macOS Rosetta 2 emulation layer coming into play? How does that work inside the VM that Docker uses behind the scenes to run Linux containers?

Reply:
Good point, I was running this on M1 which does have Rosetta 2 emulation. I wish it were easy to disable and confirm directly, but [it isn't](https://developer.apple.com/forums/thread/669486?answerId=681889022#681889022).

Alternatively, through an `x64` GitHub Codespace, I observed behavior both with and without emulation. When using an `arm64` image, calling executables indeed doesn't work until after emulation is enabled. Note the outputs as comments:

```sh
IMAGE=nextstrain/base:branch-victorlin-add-arm64-keep-scripts
docker pull --platform linux/arm64 $IMAGE

docker run $IMAGE python -c "import platform; print(platform.platform())"
# WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested
# exec /sbin/entrypoint: exec format error

docker run $IMAGE node -e "console.log(process.arch)"
# WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested
# exec /sbin/entrypoint: exec format error

# Enable emulation
docker run --rm --privileged tonistiigi/binfmt:latest --install arm64
# installing: arm64 OK

docker run $IMAGE python -c "import platform; print(platform.platform())"
# WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested
# Linux-5.4.0-1090-azure-aarch64-with-debian-10.13

docker run $IMAGE node -e "console.log(process.arch)"
# WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested
# arm64
```

So you're right, it's still up to the Docker host machine to support emulation. But I think that's something we can live with (if necessary) if the primary users of the non-`x64` images will be users of Apple silicon Macs, which have access to Rosetta 2.

Reply:
> But I think that's something we can live with (if necessary) if the primary users of the non-`x64` images will be users of Apple silicon Macs, which have access to Rosetta 2.

Agreed. Thanks for digging a bit into this above!